app/grandchallenge/algorithms/serializers.py

```diff
@@ -174,16 +171,17 @@ def __init__(self, *args, **kwargs):
             )
 
     def validate(self, data):
-        alg = data.pop("algorithm")
+        self._algorithm = data.pop("algorithm")
         user = self.context["request"].user
 
-        if not alg.active_image:
+        if not self._algorithm.active_image:
             raise serializers.ValidationError(
                 "Algorithm image is not ready to be used"
             )
         data["creator"] = user
-        data["algorithm_image"] = alg.active_image
-        data["algorithm_model"] = alg.active_model
+        data["algorithm_image"] = self._algorithm.active_image
+        data["algorithm_model"] = self._algorithm.active_model
+        data["time_limit"] = self._algorithm.time_limit
```

Comment:
Do we test to ensure that these are all fixed?

Reply:
I have added a test. These are not fields on the serializer, so they will be ignored in the user's input.